File: lib/core/storage/secure_storage.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class SecureStorage {
  static const _storage = FlutterSecureStorage();
  static const _tokenKey = 'jwt_token';
  static const _refreshTokenKey = 'jwt_refresh_token';
  static const _roleKey = 'user_role';
  static const _nameKey = 'user_name';
  static const _phoneKey = 'user_phone';
  static const _userIdKey = 'user_id';
  static const _emailKey = 'user_email';

  static Future<void> saveToken(String token) =>
      _storage.write(key: _tokenKey, value: token);

  static Future<String?> getToken() => _storage.read(key: _tokenKey);

  static Future<void> saveRefreshToken(String token) =>
      _storage.write(key: _refreshTokenKey, value: token);

  static Future<String?> getRefreshToken() =>
      _storage.read(key: _refreshTokenKey);

  static Future<void> saveRole(String role) =>
      _storage.write(key: _roleKey, value: role);

  static Future<String?> getRole() => _storage.read(key: _roleKey);

  static Future<void> saveName(String name) =>
      _storage.write(key: _nameKey, value: name);

  static Future<String?> getName() => _storage.read(key: _nameKey);

  static Future<void> savePhone(String phone) =>
      _storage.write(key: _phoneKey, value: phone);

  static Future<String?> getPhone() => _storage.read(key: _phoneKey);

  static Future<void> saveUserId(String userId) =>
      _storage.write(key: _userIdKey, value: userId);

  static Future<String?> getUserId() => _storage.read(key: _userIdKey);

  static Future<void> saveEmail(String email) =>
      _storage.write(key: _emailKey, value: email);

  static Future<String?> getEmail() => _storage.read(key: _emailKey);

  static Future<void> clearAll() => _storage.deleteAll();
}

File: src/core/storage/secureStorage.ts
import { Platform } from 'react-native';
import * as SecureStore from 'expo-secure-store';

const KEYS = {
  TOKEN: 'jwt_token',
  REFRESH_TOKEN: 'jwt_refresh_token',
  ROLE: 'user_role',
  NAME: 'user_name',
  PHONE: 'user_phone',
  USER_ID: 'user_id',
  EMAIL: 'user_email',
} as const;

const isWeb = Platform.OS === 'web';

const webStore = {
  getItemAsync: (key: string) => Promise.resolve(localStorage.getItem(key)),
  setItemAsync: (key: string, value: string) => {
    localStorage.setItem(key, value);
    return Promise.resolve();
  },
  deleteItemAsync: (key: string) => {
    localStorage.removeItem(key);
    return Promise.resolve();
  },
};

const store = isWeb ? webStore : SecureStore;

export const saveToken = (v: string) => store.setItemAsync(KEYS.TOKEN, v);
export const getToken = () => store.getItemAsync(KEYS.TOKEN);
export const saveRefreshToken = (v: string) => store.setItemAsync(KEYS.REFRESH_TOKEN, v);
export const getRefreshToken = () => store.getItemAsync(KEYS.REFRESH_TOKEN);

export const saveRole = (v: string) => store.setItemAsync(KEYS.ROLE, v);
export const getRole = () => store.getItemAsync(KEYS.ROLE);
export const saveName = (v: string) => store.setItemAsync(KEYS.NAME, v);
export const getName = () => store.getItemAsync(KEYS.NAME);
export const savePhone = (v: string) => store.setItemAsync(KEYS.PHONE, v);
export const getPhone = () => store.getItemAsync(KEYS.PHONE);
export const saveUserId = (v: string) => store.setItemAsync(KEYS.USER_ID, v);
export const getUserId = () => store.getItemAsync(KEYS.USER_ID);
export const saveEmail = (v: string) => store.setItemAsync(KEYS.EMAIL, v);
export const getEmail = () => store.getItemAsync(KEYS.EMAIL);

export const clearAll = () =>
  Promise.all(Object.values(KEYS).map((k) => store.deleteItemAsync(k)));
